crates/pepper/src/pepper_catalog.rs

```diff
@@ -0,0 +1,418 @@
+            let conn = rusqlite::Connection::open_with_flags(
+                &db_path_owned,
+                rusqlite::OpenFlags::SQLITE_OPEN_READ_WRITE
+                    | rusqlite::OpenFlags::SQLITE_OPEN_CREATE,
+            )?;
+
+            // Start transaction
+            conn.execute("BEGIN TRANSACTION", [])?;
+
+            // Get next catalog ID (for table_id)
+            let next_catalog_id: i64 = conn.query_row(
+                "SELECT value FROM pepper_metadata WHERE key = 'next_catalog_id'",
+                [],
+                |row| row.get(0),
+            )?;
+
+            let table_id = next_catalog_id;
+
+            // Generate table UUID
+            let table_uuid = uuid::Uuid::now_v7().to_string();
+
+            // Insert table metadata
+            conn.execute(
+                r"
+                INSERT INTO pepper_table (
+                    table_id, table_uuid,
+                    table_name, path, path_is_relative, schema_json, primary_key_json
+                ) VALUES (?1, ?2, ?3, ?4, ?5, ?6, ?7)
+                ",
+                rusqlite::params![
+                    table_id,
+                    table_uuid,
+                    table_name,
+                    base_path,
+                    false, // path_is_relative - using absolute paths for now
+                    schema_json,
+                    primary_key_json,
+                ],
+            )?;
+
+            // Update next_catalog_id in metadata
+            conn.execute(
+                "UPDATE pepper_metadata SET value = ?1 WHERE key = 'next_catalog_id'",
+                [next_catalog_id + 1],
+            )?;
+
+            // Commit transaction
+            conn.execute("COMMIT", [])?;
+
+            Ok::<i64, CatalogError>(table_id)
+        })
+        .await?
+    }
+
+    async fn get_table(&self, table_name: &str) -> CatalogResult<TableMetadata> {
+        let db_path_owned = self.db_path().to_string();
+        let table_name_owned = table_name.to_string();
+
+        tokio::task::spawn_blocking(move || {
+            let conn = rusqlite::Connection::open_with_flags(
+                &db_path_owned,
+                rusqlite::OpenFlags::SQLITE_OPEN_READ_ONLY,
+            )?;
+
+            // Query for the table
+            let mut stmt = conn.prepare(
+                r"
+                SELECT table_id, table_uuid,
+                       table_name, path, path_is_relative, schema_json, primary_key_json
+                FROM pepper_table
+                WHERE table_name = ?1
+                LIMIT 1
+                ",
+            )?;
+
+            let table_metadata = stmt
+                .query_row([&table_name_owned], |row| {
+                    let table_id: i64 = row.get(0)?;
+                    let table_uuid: String = row.get(1)?;
+                    let table_name: String = row.get(2)?;
+                    let path: String = row.get(3)?;
+                    let _path_is_relative: bool = row.get(4)?;
```

Comment:
```suggestion
                    let path_is_relative: bool = row.get(4)?;
```

This doesn't need to be underscored?